src/routes/(app)/post/+page.svelte

```diff
@@ -0,0 +1,246 @@
+        const params = new URLSearchParams({
+            name: query,
+            isPrivate: String(searchChannelIsPrivate),
+            page: String(channelSearchPage),
+        });
+
+        const res = await fetch(`api/s/channels?${params}`);
+        if (!res.ok) {
+            console.error(await res.text());
+            channelSearchError = 'failed to search';
+            return [];
+        } else {
+            return await res.json();
+        }
+    };
+
+    const loadMoreChannels = async () => {
+        channelSearchPage++;
+        const more = await searchChannels(channelSearchQuery);
+        channels = channels.concat(more);
+    };
+
+    const selectChannel = (channel: MiniChannel) => {
+        selectedChannel = channel;
+        resetState();
+    };
+
+    const uploadVideo = async (event: SubmitEvent) => {
+        if (!data.uploadKey || !data.expirationTime || !data.videoId) return;
+        event.preventDefault();
+        if (!event.target) throw new Error('Something went wrong');
+        const formData = new FormData(event.target as HTMLFormElement);
+        const videoFile = formData.get('video') as File;
+        if (!videoFile) {
+            toast.error('Please submit video');
+            console.error("Something went wrong submitting video. Ensure you've provided one.");
+            return;
+        }
+
+        upload = new tus.Upload(videoFile as File, {
+            endpoint: 'https://video.bunnycdn.com/tusupload',
+            retryDelays: [0, 3000, 5000, 10000, 20000, 60000, 60000],
+            headers: {
+                AuthorizationSignature: data.uploadKey,
+                AuthorizationExpire: data.expirationTime.toString(),
+                VideoId: data.videoId,
+                LibraryId: PUBLIC_LIBRARY_ID,
+            },
+            metadata: {
+                filetype: videoFile.type,
+                title: data.post.id,
+            },
+            onError: (error) => {
+                toast.error(`upload failed (${error.message}). Retrying...`);
+                console.error(error);
+            },
+            onProgress: (bytesUploaded, bytesTotal) => {
+                uploadProgress = (bytesUploaded / bytesTotal) * 100;
+            },
+            onSuccess: async () => {
+                // update post to be status=OK
+                const res = await fetch(`/api/post/${data.post.id}`, {
+                    method: 'PUT',
+                    body: JSON.stringify({ status: 'OK' }),
+                });
+
+                if (!res.ok) {
+                    toast.error('Something went wrong updating post status');
+                    console.error('Something went wrong updating post status');
+                }
+
+                uploadProgress = 100;
+                toast.success('Video uploaded!');
+
+                formState = 'COMPLETE';
+            },
+        });
+        upload.start();
+        uploadProgress = 0;
```

Comment:
Might be nicer to have this be in a utility somewhere else and have it return an object with a store for progress and a promise that either succeeds or errors

Reply:
Yeah I was gonna suggest as much. I'd wanna put some thought into an abstraction for this.

Reply:
Sounds good.  I've added a story to sprint 3 for this.